crates/pubsub/src/implementors/mosquitto.rs

```diff
@@ -0,0 +1,157 @@
+use std::sync::{Arc, Mutex};
+
+use anyhow::{Context, Result};
+use async_channel::Receiver;
+use futures::executor::block_on;
+use mosquitto_rs::{Client, Message, QoS};
+use slight_common::BasicState;
+
+#[derive(Clone)]
+pub struct MosquittoImplementor {
+    mqtt: Arc<Mutex<Client>>,
```

Comment:
I would be careful about using `Arc<Mutex<...>>` because it has a performance overhead. In what scenerios did you see the necessity of `Arc<Mutex<...>>` on `mqtt`, `subscriptions` and `subscriber`?

Reply:
That's right~ iirc, `subscriber` needed a `&mut self`, so I had to `Arc<Mutex<>>` it.

Reply:
That's very restricted, isn't it? I wonder why wit-bindgen doesn't give us a `&mut self_` for modifying the resources.